connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -1,132 +1,253 @@
-        // environments that control classloading differently (OSGi, Spring and others) and don't
-        // depend on the System classloader to load Connect's classes.
-        this(DelegatingClassLoader.class.getClassLoader());
-    }
-
-    /**
-     * Retrieve the PluginClassLoader associated with a plugin class
-     * @param name The fully qualified class name of the plugin
-     * @return the PluginClassLoader that should be used to load this, or null if the plugin is not isolated.
-     */
-    // VisibleForTesting
-    PluginClassLoader pluginClassLoader(String name) {
-        if (!PluginUtils.shouldLoadInIsolation(name)) {
-            return null;
-        }
-        SortedMap<PluginDesc<?>, ClassLoader> inner = pluginLoaders.get(name);
-        if (inner == null) {
-            return null;
-        }
-        ClassLoader pluginLoader = inner.get(inner.lastKey());
-        return pluginLoader instanceof PluginClassLoader
-               ? (PluginClassLoader) pluginLoader
-               : null;
-    }
-
-    ClassLoader connectorLoader(String connectorClassOrAlias) {
-        String fullName = aliases.getOrDefault(connectorClassOrAlias, connectorClassOrAlias);
-        ClassLoader classLoader = pluginClassLoader(fullName);
-        if (classLoader == null) classLoader = this;
-        log.debug(
-            "Getting plugin class loader: '{}' for connector: {}",
-            classLoader,
-            connectorClassOrAlias
-        );
-        return classLoader;
-    }
-
-    public void installDiscoveredPlugins(PluginScanResult scanResult) {
-        pluginLoaders.putAll(computePluginLoaders(scanResult));
-        for (String pluginClassName : pluginLoaders.keySet()) {
-            log.info("Added plugin '{}'", pluginClassName);
-        }
-        aliases.putAll(PluginUtils.computeAliases(scanResult));
-        for (Map.Entry<String, String> alias : aliases.entrySet()) {
-            log.info("Added alias '{}' to plugin '{}'", alias.getKey(), alias.getValue());
-        }
-    }
-
-    @Override
-    protected Class<?> loadClass(String name, boolean resolve) throws ClassNotFoundException {
-        String fullName = aliases.getOrDefault(name, name);
-        PluginClassLoader pluginLoader = pluginClassLoader(fullName);
-        if (pluginLoader != null) {
-            log.trace("Retrieving loaded class '{}' from '{}'", fullName, pluginLoader);
-            return pluginLoader.loadClass(fullName, resolve);
-        }
-
-        return super.loadClass(fullName, resolve);
-    }
-
-    private static Map<String, SortedMap<PluginDesc<?>, ClassLoader>> computePluginLoaders(PluginScanResult plugins) {
-        Map<String, SortedMap<PluginDesc<?>, ClassLoader>> pluginLoaders = new HashMap<>();
-        plugins.forEach(pluginDesc ->
-                pluginLoaders.computeIfAbsent(pluginDesc.className(), k -> new TreeMap<>())
-                        .put(pluginDesc, pluginDesc.loader()));
-        return pluginLoaders;
-    }
-}
+/*

```

Comment:
nit: is there a big whitespace change in this class? The diff looks huge.

Reply:
I'm not sure why this is the case tbh. Maybe just the magnitude of changes in the class is making the diff look so big 🤔